include/cuda/std/detail/libcxx/include/support/atomic/atomic_cuda.h

```diff
@@ -121,13 +121,43 @@ const volatile _Tp* __cxx_get_underlying_device_atomic(__cxx_atomic_base_heterog
 }
 
 template <typename _Tp>
-using __cxx_atomic_small_to_32 = typename conditional<is_signed<_Tp>::value, int32_t, uint32_t>::type;
+using __cxx_atomic_small_to_32 =
+    typename conditional<is_signed<_Tp>::value,
+        int32_t,
+        uint32_t
+    >::type;
+
+// Arithmetic conversions to/from proxy types
+template<class _Tp, _EnableIf<is_arithmetic<_Tp>::value, int> = 0>
```

Comment:
Ditto, we should move to
```suggestion
template<class _Tp, __enable_if_t<is_arithmetic<_Tp>::value, int> = 0>
```

Reply:
I was under the impression that the meta base stuff was better to use internally.

Reply:
That is correct, but my evil plan is to settle on `__enable_if_t` globally and then move that over to use the meta based implementation. I really do not like it that we have three different ways to do this.

So anything here is fine by me as that should be a separate PR anyway

Reply:
That is really strange, do we build in C++03?

Reply:
I rebased the branch, which I know is fairly aggressive, but now all the conditionals are using the `__blah_t` versions.